### ISSUE TITLE
bug: binary type conversion issue with memtable registration

### ISSUE THREAD

**Issue by ededovic** [ibis-project/ibis#5810]
### What happened?

Here is a different minimal example. Workaround is to convert 'UNK' into null or the entire column into string. 

```
    zc = pd.Series([ 6770, 54601, 54650, 37043, 16146,'UNK', 1550, 42211, 15601, 1440], name='zip')
    zcv = ibis.memtable(zc)
    zcv.schema()
```
comes back with: 
```
ibis.Schema {
  zip  binary
}
```

but trying to show just what's in the table `zcv` shows the below error: 
```
Traceback (most recent call last):
  File "...\envs\ml310\lib\site-packages\IPython\core\formatters.py", line 221, in catch_format_error
    r = method(self, *args, **kwargs)
  File "...\envs\ml310\lib\site-packages\IPython\core\formatters.py", line 706, in __call__
    printer.pretty(obj)
  File "...\envs\ml310\lib\site-packages\IPython\lib\pretty.py", line 410, in pretty
    return _repr_pprint(obj, self, cycle)
  File "...\envs\ml310\lib\site-packages\IPython\lib\pretty.py", line 778, in _repr_pprint
    output = repr(obj)
  File "...\envs\ml310\lib\site-packages\ibis\expr\types\core.py", line 56, in __repr__
    simple_console.print(self)
  File "...\envs\ml310\lib\site-packages\rich\console.py", line 1698, in print
    extend(render(renderable, render_options))
  File "...\envs\ml310\lib\site-packages\rich\console.py", line 1310, in render
    render_iterable = renderable.__rich_console__(self, _options)  # type: ignore[union-attr]
  File "...\envs\ml310\lib\site-packages\ibis\expr\types\relations.py", line 259, in __rich_console__
    table = to_rich_table(self, width)
  File "...\envs\ml310\lib\site-packages\ibis\expr\types\pretty.py", line 230, in to_rich_table
    result = table.limit(nrows + 1).execute()
  File "...\envs\ml310\lib\site-packages\ibis\expr\types\core.py", line 303, in execute
    return self._find_backend(use_default=True).execute(
  File "...\envs\ml310\lib\site-packages\ibis\backends\base\sql\__init__.py", line 246, in execute
    self._register_in_memory_tables(expr)
  File "...\envs\ml310\lib\site-packages\ibis\backends\base\sql\__init__.py", line 262, in _register_in_memory_tables
    self._register_in_memory_table(memtable)
  File "...\envs\ml310\lib\site-packages\ibis\backends\duckdb\__init__.py", line 695, in _register_in_memory_table
    table = op.data.to_pyarrow(schema)
  File "...\envs\ml310\lib\site-packages\ibis\backends\pandas\client.py", line 236, in to_pyarrow
    return pa.Table.from_pandas(self._df, schema=ibis_to_pyarrow_schema(schema))
  File "pyarrow\table.pxi", line 3557, in pyarrow.lib.Table.from_pandas
  File "...\envs\ml310\lib\site-packages\pyarrow\pandas_compat.py", line 611, in dataframe_to_arrays
    arrays = [convert_column(c, f)
  File "...\envs\ml310\lib\site-packages\pyarrow\pandas_compat.py", line 611, in <listcomp>
    arrays = [convert_column(c, f)
  File "...\envs\ml310\lib\site-packages\pyarrow\pandas_compat.py", line 598, in convert_column
    raise e
  File "...\envs\ml310\lib\site-packages\pyarrow\pandas_compat.py", line 592, in convert_column
    result = pa.array(col, type=type_, from_pandas=True, safe=safe)
  File "pyarrow\array.pxi", line 316, in pyarrow.lib.array
  File "pyarrow\array.pxi", line 83, in pyarrow.lib._ndarray_to_array
  File "pyarrow\error.pxi", line 123, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: ("Expected bytes, got a 'int' object", 'Conversion failed for column zip with type object')

```

### What version of ibis are you using?

5.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
@ededovic What would you like to happen here?

Possibilities of the top of my head:

1. Raise an error
2. Use a `null` type (as a stand-in for an unknown type)

**Comment by ededovic** [comment]
Thank you. Good point. My thinking is that if `ibis.memtable` took it as binary then subsequent functions should not be complaining and trying to convert it to something else. I would either stop arrow's auto conversion or do it earlier.  
If the conversion needs to happen, either I do it or `ibis.memtable` complains about it and ask me what to do, but not subsequent functions. That makes it easier to debug issues. 

I can see instances when the column will truly be binary, and ibis should not fail because it is an allowed datatype.

**Comment by cpcloud** [comment]
Fixed by #5853. Will be release in 6.0, (release after next). The fix is merged into the `6.x.x` branch.
